sympy/integrals/transforms.py

```diff
@@ -1137,6 +1160,541 @@ def sbs(expr):
+                    debug(' ---> go')
+                    return _laplace_cr(k*s_dom.xreplace(ma),
+                                plane.xreplace(ma), S.true, **hints)
+            except Exception:
+                debug('_laplace_apply_rules did not match.')
+    # else:
+    if f.has(DiracDelta):
+        raise IntegralTransformError(
+                    'Laplace', None, 'No rules could be applied.')
+    ma1 = func.match(g)
+    if ma1:
+        arg = ma1[g].args[0].collect(t)
+        ma2 = arg.match(b*t)
+        if ma2 and ma2[b]>0:
+            debug('_laplace_apply_rules match:')
+            debug('      f:    %s ( %s, %s )'%(f, ma1, ma2))
+            debug('      rule: amplitude and time scaling (1.1, 1.2)')
+            if ma2[b]==1:
+                if doit==True and not any(func.has(t) for func
+                                          in ma1[g].atoms(AppliedUndef)):
+                    return k*_laplace_transform(ma1[g].func(t), t, s, **hints)
+                else:
+                    return k*LaplaceTransform(ma1[g].func(t), t, s, **hints)
+            else:
+                L = _laplace_apply_rules(ma1[g].func(t), t, s/ma2[b], **hints)
+                try:
+                    r, p, c = L
+                    return (k/ma2[b]*r, p, c)
+                except TypeError:
+                    return k/ma2[b]*L
+    ma1 = func.match(Heaviside(y)*g)
+    if ma1:
+        ma2 = ma1[y].match(t-a)
+        ma3 = ma1[g].args[0].collect(t).match(t-b)
+        if ma2 and ma2[a]>0 and ma3 and ma2[a]==ma3[b]:
+            debug('_laplace_apply_rules match:')
+            debug('      f:    %s ( %s, %s, %s )'%(f, ma1, ma2, ma3))
+            debug('      rule: time shift (1.3)')
+            L = _laplace_apply_rules(ma1[g].func(t), t, s, **hints)
+            try:
+                r, p, c = L
+                return (k*exp(-ma2[a]*s)*r, p, c)
+            except TypeError:
+                return k*exp(-ma2[a]*s)*L
+    ma1 = func.match(exp(y)*z)
+    if ma1:
+        ma2 = ma1[y].collect(t).match(a*t)
+        if ma2:
+            debug('_laplace_apply_rules match:')
+            debug('      f:    %s ( %s, %s )'%(f, ma1, ma2))
+            debug('      rule: multiply with exp (1.4)')
+            L = _laplace_apply_rules(ma1[z], t, s-ma2[a], **hints)
+            try:
+                r, p, c = L
+                return (r, p+ma2[a], c)
+            except TypeError:
+                return L
+    ma1 = func.match(sinh(y)*z)
+    if ma1:
+        ma2 = ma1[y].collect(t).match(a*t)
+        if ma2:
+            debug('_laplace_apply_rules match:')
+            debug('      f:    %s ( %s, %s )'%(f, ma1, ma2))
+            debug('      rule: multiply with sinh (1.5)')
+            L = _laplace_apply_rules(ma1[z], t, s, **hints)
+            try:
+                r, p, c = L
+                return ((r.subs(s, s-ma2[a])-r.subs(s, s+ma2[a])).simplify()/2,
+                        p+Abs(ma2[a]), c)
+            except TypeError:
```

Comment:
Why is this catching TypeError? What raises the error?

In general it is better to avoid catching TypeError because it risks masking bugs. There are many situations where a bug/typo etc can lead to a TypeError but this would suppress the traceback.

Reply:
The line `r, p, c = L` raises a TypeError if the unpacking fails. `L = _laplace_apply_rules(ma1[z], t, s, **hints)` above can return a tuple or a `LaplaceTransform` object.  This was my attempt to write "if it is a tuple, do it on the first entry, otherwise do it on everything"

Reply:
Is there a need for `_laplace_apply_rules` to return ` LaplaceTransform` object? Does that just represent a failure to compute the Laplace transform? If so then it could just yse `return None` instead and then:
```python
L = f()
if L is None:
   ...
```

Reply:
> Is there a need for `_laplace_apply_rules` to return `LaplaceTransform` object?

Oh yes, `_laplace_apply_rules` must return `LaplaceTransform` objects when it simplifies expressions with undefined functions, see e.g. the `diff` stuff in https://github.com/sympy/sympy/pull/22376#issuecomment-971412783 .

Reply:
Should it not still return a tuple in that case (with the convergence conditions)?

Reply:
There is no convergence condition for undefined functions, so I think it should not return any in that case. 

What could, in principle, be done is to implement a "convergence plane modifier", so, e.g., `sinh(a*t)*f(t)` would mean that the correct convergencer plane is the maximum of `Abs(a)` and the convergence plane of transforming `f(t)` on its own, but this would be a major hassle to implement.

I also think it is not necessary. Convergence in the Laplace Transform just means "right of the right-most singularity", so I think it is OK if determining the convergence condition after substituting a concrete function for an undefined function in the result is left as an exercise for the user. If they want to have an automatic convergence plane, they shall use defined functions ;)

Reply:
I just mean that it's a bit weird to have a function that sometimes returns a tuple and sometimes returns a LaplaceTransform. Looking at it now I realise that's what `laplace_transform` does by default:
```python
In [7]: laplace_transform(f(t), t, s)
Out[7]: LaplaceTransform(f(t), t, s)

In [8]: laplace_transform(exp(-t), t, s)
Out[8]: 
⎛  1           ⎞
⎜─────, 0, True⎟
⎝s + 1         ⎠
```
That's a bit unfortunate...

Reply:
It is unfortunate, yes!  I think we should leave it as is because changing this will affect a lot of users.